From mboxrd@z Thu Jan  1 00:00:00 1970
Return-Path: <amd-gfx-bounces@lists.freedesktop.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from gabe.freedesktop.org (gabe.freedesktop.org [131.252.210.177])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id BC2F4C369C1
	for <amd-gfx@archiver.kernel.org>; Tue, 15 Apr 2025 14:59:53 +0000 (UTC)
Received: from gabe.freedesktop.org (localhost [127.0.0.1])
	by gabe.freedesktop.org (Postfix) with ESMTP id 5BA2310E7BF;
	Tue, 15 Apr 2025 14:59:53 +0000 (UTC)
Authentication-Results: gabe.freedesktop.org;
	dkim=pass (1024-bit key; unprotected) header.d=amd.com header.i=@amd.com header.b="GlKYfoVJ";
	dkim-atps=neutral
Received: from NAM10-DM6-obe.outbound.protection.outlook.com
 (mail-dm6nam10on2062.outbound.protection.outlook.com [40.107.93.62])
 by gabe.freedesktop.org (Postfix) with ESMTPS id E3EB910E7C1
 for <amd-gfx@lists.freedesktop.org>; Tue, 15 Apr 2025 14:59:51 +0000 (UTC)
ARC-Seal: i=1; a=rsa-sha256; s=arcselector10001; d=microsoft.com; cv=none;
 b=EBhZCYpJRQGa9Xwkt+OcO7rKyndWWaTckpiv13HhIBG4U+JK/B4lnFi55lP9kREDH2k4roOpJR5ehsFJ0FJ0DRRWIO0BSekNQs48C/SFiyabTR0zyn1X4yQ9Eofh65yEHDwLnHJA1QiTnzCF5pdHQ4but2QxZRIDM+F21GFh/Wk/GeTdUl8fZW01f5s80wN5GC9hPxd9d09pLTKHdZxCC5PX6zyb0Bb1BFxfz/rVoAF4HjxLfxbLol6ifdWi5jOxk+3m8Zy5yjckJuU0J3bD5KJjLB8pDPO8SwCLO1tSsVXBuw3yPew82fsQgJIx+1CaKn/NSW0f40wcvUeajmfuow==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com; 
 s=arcselector10001;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=9/xZ3Jm5Qhzs1lRjm76lJGEB6XyewX15AbybibjjXGc=;
 b=rUnNAl3s6H2pdXg+GqPlwrfaUO8NqTQiNHZNq+qmJeTD64Ps20iJvsaafOP3DUc3AUmmm7FuRqjUzQGHIZpZbxpKp8aJrUoMPE1V7l6AzMzud+oeYItspekbddCZuTArpl9L+/ztNRBvgX84eX0FfuPxNJeof8DjIMmILdhMZKovrHA5noQYi4Z2YfLYJNNhjVsXwM/QQCzr50GphRxjI6M7Bg38iOaZ0n9SjfZwJVohcFsqvTZch6M4YlPUviSCHqKcZLFp2O9pcEPxVuw/lj9n6f3/Z4jYq32q3ysM7OYQ2g+dj4MP5HvpHAzlZToq3aHuPpcNMDZrQOdWCrufZQ==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass (sender ip is
 165.204.84.17) smtp.rcpttodomain=lists.freedesktop.org smtp.mailfrom=amd.com; 
 dmarc=pass (p=quarantine sp=quarantine pct=100) action=none
 header.from=amd.com; dkim=none (message not signed); arc=none (0)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=amd.com; s=selector1; 
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=9/xZ3Jm5Qhzs1lRjm76lJGEB6XyewX15AbybibjjXGc=;
 b=GlKYfoVJylIhiLomPTRMLidSNR7P4IOtkb5QDQcEHbwAvMz4zbnNSG4Te8Jzx6cjSy5cJi8A9nIyQFlERt4Gqw0o8vufVB/GiFLdsYgU5lfQFhPi/tpqFaIQn29XxLcY8rFIhfumZUKzoqwHohtVEEXDUPdTZ8AyP3yH4cbcf2E=
Received: from PH8P221CA0050.NAMP221.PROD.OUTLOOK.COM (2603:10b6:510:346::11)
 by DM3PR12MB9351.namprd12.prod.outlook.com (2603:10b6:8:1ac::9) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.8632.35; Tue, 15 Apr
 2025 14:59:44 +0000
Received: from SA2PEPF000015CC.namprd03.prod.outlook.com
 (2603:10b6:510:346:cafe::38) by PH8P221CA0050.outlook.office365.com
 (2603:10b6:510:346::11) with Microsoft SMTP Server (version=TLS1_3,
 cipher=TLS_AES_256_GCM_SHA384) id 15.20.8632.36 via Frontend Transport; Tue,
 15 Apr 2025 14:59:43 +0000
X-MS-Exchange-Authentication-Results: spf=pass (sender IP is 165.204.84.17)
 smtp.mailfrom=amd.com; dkim=none (message not signed)
 header.d=none;dmarc=pass action=none header.from=amd.com;
Received-SPF: Pass (protection.outlook.com: domain of amd.com designates
 165.204.84.17 as permitted sender) receiver=protection.outlook.com;
 client-ip=165.204.84.17; helo=SATLEXMB04.amd.com; pr=C
Received: from SATLEXMB04.amd.com (165.204.84.17) by
 SA2PEPF000015CC.mail.protection.outlook.com (10.167.241.202) with Microsoft
 SMTP Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256) id
 15.20.8655.12 via Frontend Transport; Tue, 15 Apr 2025 14:59:43 +0000
Received: from mkmmarleung05.amd.com (10.180.168.240) by SATLEXMB04.amd.com
 (10.181.40.145) with Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256) id 15.1.2507.39; Tue, 15 Apr
 2025 09:59:41 -0500
From: Zaeem Mohamed <zaeem.mohamed@amd.com>
To: <amd-gfx@lists.freedesktop.org>
CC: Harry Wentland <harry.wentland@amd.com>, Leo Li <sunpeng.li@amd.com>,
 Aurabindo Pillai <aurabindo.pillai@amd.com>, Roman Li <roman.li@amd.com>,
 Wayne Lin <wayne.lin@amd.com>, Tom Chung <chiahsuan.chung@amd.com>, "Fangzhi
 Zuo" <jerry.zuo@amd.com>, Zaeem Mohamed <zaeem.mohamed@amd.com>, "Daniel
 Wheeler" <daniel.wheeler@amd.com>, Alex Hung <alex.hung@amd.com>, Roman Li
 <Roman.Li@amd.com>, <stable@vger.kernel.org>, Mario Limonciello
 <mario.limonciello@amd.com>
Subject: [PATCH 07/22] drm/amd/display: Force full update in gpu reset
Date: Tue, 15 Apr 2025 10:58:53 -0400
Message-ID: <20250415145908.3790253-8-zaeem.mohamed@amd.com>
X-Mailer: git-send-email 2.34.1
In-Reply-To: <20250415145908.3790253-1-zaeem.mohamed@amd.com>
References: <20250415145908.3790253-1-zaeem.mohamed@amd.com>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
Content-Type: text/plain
X-Originating-IP: [10.180.168.240]
X-ClientProxiedBy: SATLEXMB03.amd.com (10.181.40.144) To SATLEXMB04.amd.com
 (10.181.40.145)
X-EOPAttributedMessage: 0
X-MS-PublicTrafficType: Email
X-MS-TrafficTypeDiagnostic: SA2PEPF000015CC:EE_|DM3PR12MB9351:EE_
X-MS-Office365-Filtering-Correlation-Id: d249bb14-d006-41c1-5d37-08dd7c2e27ad
X-MS-Exchange-SenderADCheck: 1
X-MS-Exchange-AntiSpam-Relay: 0
X-Microsoft-Antispam: BCL:0;
 ARA:13230040|376014|1800799024|82310400026|36860700013; 
X-Microsoft-Antispam-Message-Info: =?us-ascii?Q?8oGxK2YRHBpqVluII6lhoh/+MszKWo6xTmTWw7Z6A/hemfEiUFdr3S06bYMc?=
 =?us-ascii?Q?N4OgIjksWPsF/udjm/RLn63I+CRpWkOEVw+MAAJlXH+1m8dDpO6zZebRZ8kp?=
 =?us-ascii?Q?QAkiwSdPEbyaCWSpPsQWapUt/iIMvD9BcEzEeFs1HguYArAB5hlUJZgHNhg1?=
 =?us-ascii?Q?8DcSgbgdIWk28draYYZ1LYjMeSU3KM1B2pWkC2as/Ts3vJ2WfOTRc1cshHUD?=
 =?us-ascii?Q?w7e1i52acVk42is8SLr38TNKzctZ/Svv/2fRzh/oWKsFaN0bM7AG/lBet3ND?=
 =?us-ascii?Q?5VK+14lK2O0g1X1uTeS/MqlyFHyDoBuPMU61pzmLi/tSd/QmzL6M1ErmAyob?=
 =?us-ascii?Q?XbmIX3te4WBAzjzlPvzxLOf69L4jYLM7JYcvWRe6MZxHWJLRozKABIRw9Li/?=
 =?us-ascii?Q?pXTo/zbAcrjsk/vnMsXKmdh/XLGM/c/9DhrLpVPoHN4kCAaTRPzqaaoyZoL8?=
 =?us-ascii?Q?mGetZUVEMIKWqpzaoMqwdAOByBUPB6tO6Rdf+om9umjf5N4Yb76kJxNWb/9s?=
 =?us-ascii?Q?XCVoPf8OPCvFPe/dTCogHo5YIARTb3JLeR2YRNzW/rXGzV6MroK/YXFGaM49?=
 =?us-ascii?Q?mzyYUwaJipZn/RnkplXRoYa+8KA0/Wc3krsiiITDHt/kcLkEsF9YHa8cSa+N?=
 =?us-ascii?Q?TeBrXtVt2xTz0WC4Wv8jqnY1zmX0tICj8rj0CsCq/V4x5puODznkOyfXaC4w?=
 =?us-ascii?Q?Ni6wOPwNZisIy45qazEP/79NxVNf8tLEcWK66S714pK2BEUpS1M18ZjS2A96?=
 =?us-ascii?Q?u9iRHSHgYwzAs+g9emoN6mWJio2ckIGcH7w2zlVU2UYFiH663CdQQOtAsxkI?=
 =?us-ascii?Q?CwcXWWeOKMgIst3FqxT9JV1jjruG7ESbbRKU+VzvVj9hnMLWxVv7YpzfvBNH?=
 =?us-ascii?Q?674X7Ab444XMBgeisLRsCygDK+LZjxCJ0ENb0eam5aA9wZHXQrnGIb8trJMY?=
 =?us-ascii?Q?IiDeXT8ugxz1FNkpjhscHmX6xh0zOA+TkoSiyxmtxuxDLuGHQ6syvWlng8De?=
 =?us-ascii?Q?oc6B3n99clu4mAAZIl0zV5IY3INFbQ3LpE3H82tLprpVhjkEk0rn2WFOFivy?=
 =?us-ascii?Q?HlM9TmOBNykNJywDCaEJNoGdlgHvOC3F+TLTYiSi0aEXfP3W4O/yjEYqCVPX?=
 =?us-ascii?Q?TVAc+riieTAlgjdpsKFcBJ3zwh4ugi1p0Jjz31a1QUDfoVRXcgDWDFlTPxah?=
 =?us-ascii?Q?p5paKwr+Z38yNIvU5oi0gSpFDsDiERpoW+ImMbhdhLwkT+pNaanEOOExye9S?=
 =?us-ascii?Q?Fl10Rwx/gLUMFwYZ3PqSD0shLLw5W/y3j3cE78kKWCs4HNLhIxN0/TRWSMkc?=
 =?us-ascii?Q?OnfJabv5xPhyY8TfKLd8ESS2YbDZyC8jjxjoKja1Y0Y0vP4Lr8T41JxjoKZg?=
 =?us-ascii?Q?SX5eAwJrMIBM4QHtgTWe5BMJl5UEPuIVx+DQE2liH1prsNT0XPFw/Oqe9o+c?=
 =?us-ascii?Q?IZIDLAE9/1ucGKdbaaRb/9wUzOvrcquOTcrebJrzMG/EE1UQGbjnV2hoI3P9?=
 =?us-ascii?Q?vr12J0V1hgqQaNJSmjTcOfLsPJpiBB6hiTPp?=
X-Forefront-Antispam-Report: CIP:165.204.84.17; CTRY:US; LANG:en; SCL:1; SRV:;
 IPV:CAL; SFV:NSPM; H:SATLEXMB04.amd.com; PTR:InfoDomainNonexistent; CAT:NONE;
 SFS:(13230040)(376014)(1800799024)(82310400026)(36860700013); DIR:OUT;
 SFP:1101; 
X-OriginatorOrg: amd.com
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 15 Apr 2025 14:59:43.1866 (UTC)
X-MS-Exchange-CrossTenant-Network-Message-Id: d249bb14-d006-41c1-5d37-08dd7c2e27ad
X-MS-Exchange-CrossTenant-Id: 3dd8961f-e488-4e60-8e11-a82d994e183d
X-MS-Exchange-CrossTenant-OriginalAttributedTenantConnectingIp: TenantId=3dd8961f-e488-4e60-8e11-a82d994e183d; Ip=[165.204.84.17];
 Helo=[SATLEXMB04.amd.com]
X-MS-Exchange-CrossTenant-AuthSource: SA2PEPF000015CC.namprd03.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Anonymous
X-MS-Exchange-CrossTenant-FromEntityHeader: HybridOnPrem
X-MS-Exchange-Transport-CrossTenantHeadersStamped: DM3PR12MB9351
X-BeenThere: amd-gfx@lists.freedesktop.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: Discussion list for AMD gfx <amd-gfx.lists.freedesktop.org>
List-Unsubscribe: <https://lists.freedesktop.org/mailman/options/amd-gfx>,
 <mailto:amd-gfx-request@lists.freedesktop.org?subject=unsubscribe>
List-Archive: <https://lists.freedesktop.org/archives/amd-gfx>
List-Post: <mailto:amd-gfx@lists.freedesktop.org>
List-Help: <mailto:amd-gfx-request@lists.freedesktop.org?subject=help>
List-Subscribe: <https://lists.freedesktop.org/mailman/listinfo/amd-gfx>,
 <mailto:amd-gfx-request@lists.freedesktop.org?subject=subscribe>
Errors-To: amd-gfx-bounces@lists.freedesktop.org
Sender: "amd-gfx" <amd-gfx-bounces@lists.freedesktop.org>

From: Roman Li <Roman.Li@amd.com>

[Why]
While system undergoing gpu reset always do full update
to sync the dc state before and after reset.

[How]
Return true in should_reset_plane() if gpu reset detected

Cc: <stable@vger.kernel.org>

Reviewed-by: Aurabindo Pillai <aurabindo.pillai@amd.com>
Reviewed-by: Mario Limonciello <mario.limonciello@amd.com>
Signed-off-by: Roman Li <Roman.Li@amd.com>
Signed-off-by: Zaeem Mohamed <zaeem.mohamed@amd.com>
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
index dace1e42f412..46e0de6cc277 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -11122,6 +11122,9 @@ static bool should_reset_plane(struct drm_atomic_state *state,
 	    state->allow_modeset)
 		return true;
 
+	if (amdgpu_in_reset(adev) && state->allow_modeset)
+		return true;
+
 	/* Exit early if we know that we're adding or removing the plane. */
 	if (old_plane_state->crtc != new_plane_state->crtc)
 		return true;
-- 
2.34.1


